### PR TITLE
[JENKINS-33266] Upgrade to new Parent POM

### PR DESCRIPTION
**Pull request by andresrc**
* [x] Upgrade to parent POM 2.3
* [x] Migrate test cases from `HudsonTestCase` to `JenkinsRule`
* [x] Fix findbugs errors when building against 1.642.1 (PCT scenario).

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -80,66 +73,7 @@
...
-            <phase>verify</phase> 
-            <goals>
-              <goal>check</goal> 
-            </goals>
-          </execution>
-        </executions>
-      </plugin>
-      <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>animal-sniffer-maven-plugin</artifactId>
```
Comment (oleg-nenashev) [inline pom.xml]: :bug: the new parent pom does not explicitly call animal sniffer

Comment (andresrc) [inline pom.xml]: @oleg-nenashev, it is included by default:

https://github.com/jenkinsci/plugin-pom/blob/master/pom.xml#L560

On a `mvn clean install`:

```
[INFO] --- animal-sniffer-maven-plugin:1.14:check (check) @ mailer ---
[INFO] Resolved signature org.codehaus.mojo.signature:java16 version as 1.1 from dependencyManagement
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java16:1.1
```

Comment (oleg-nenashev) [inline pom.xml]: Uhh, sorry. I've missed it

File: pom.xml [hunk 2]
```diff
@@ -21,18 +21,11 @@
   </licenses>
   <properties>
     <powermock.version>1.6.2</powermock.version>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (lordofthejars) [inline pom.xml]: I think it is a good idea to have a project defining the encoding.

**Comment by andresrc** [comment]
CI Builds are getting stuck with no apparent reason. Builds verified locally.

File: pom.xml (hunk 2 above)
Comment (andresrc) [inline pom.xml]: I agree, that's why I maintain the default value defined in the parent pom.

**Comment by armfergom** [comment]
LGTM :bee:

File: src/test/java/hudson/tasks/MailerTest.java [hunk 3]
```diff
@@ -37,12 +40,21 @@
...
 /**
  * @author Kohsuke Kawaguchi
  */
-public class MailerTest extends HudsonTestCase {
+public class MailerTest {
+
+    @Rule
+    public JenkinsRule rule = new JenkinsRule();
+
     @Bug(1566)
```
Comment (jglick) [inline src/test/java/hudson/tasks/MailerTest.java]: BTW can use `@Issue` now.

**Comment by jglick** [comment]
:bee:

File: src/test/java/hudson/tasks/MailerTest.java (hunk 3 above)
Comment (andresrc) [inline src/test/java/hudson/tasks/MailerTest.java]: When migrating from `Bug` to `Issue` is some "marker" used to "decorate" the previous value or the plain number represents the old bug tracking system?

Comment (olivergondza) [inline src/test/java/hudson/tasks/MailerTest.java]: `@Bug(42)` should be converted to `@Issue("JENKINS-42")`, I do not see any need for additional marking.

**Pull request merged**